src/train.py

```diff
@@ -101,18 +137,44 @@ def main(argv):
             labels = [CLASS_INDEX[label] for label in labels]
             labels = torch.tensor(labels).to(DEVICE)
 
-            # Zero the parameter gradients
-            optimizer.zero_grad()
+            style_vector = torch.rand((BATCH_SIZE, 100), device=DEVICE)
+            char_vec = char_vector(BATCH_SIZE, DEVICE)
+
+            # Generate characters
+            chars_generated = gen(char_vec, style_vector)
+
+            # Zero the parameter gradients for the discriminator
+            optimizer_disc.zero_grad()
+
+            # Make predictions
+            prediction_real = disc(font)
+            prediction_fake = disc(chars_generated)
+            prediction_combined = torch.cat((prediction_real, prediction_fake),
+                                            0)
+
+            # combine char vector from real and fake CHARACTERS
+            labels_combined = torch.cat((labels, char_vec), 0)
```

Comment:
This should not be with `char_vec`. The discriminator needs to know that the fake is fake, not what they are intended to be.